capstone/src/main/java/com/google/sps/servlets/YoutubeServlet.java

```diff
@@ -38,20 +38,27 @@
   private static final String KEY = "API_KEY";
   private static final String NUM_RESULTS = "5";
   private static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
+  private String postRequestBodyData;
+  private String channelId;
+  private String videoId; 
   OkHttpClient client = new OkHttpClient();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String ChannelId = request.getParameter("channelId");
-    String videoId = request.getParameter("videoId");
-    String completeUrl = (ChannelId != null) ? 
-      (BASE_URL + "&allThreadsRelatedToChannelId=" + ChannelId + "&maxResults=" + NUM_RESULTS + "&key=" + KEY) : 
-        (BASE_URL + "&videoId=" + videoId + "&maxResults=" + NUM_RESULTS + "&key=" + KEY);
+    String completeUrl = (postRequestBodyData.length() == 24) ? 
+      (BASE_URL + "&allThreadsRelatedToChannelId=" + postRequestBodyData + "&maxResults=" + NUM_RESULTS + "&key=" + KEY) : 
+        (BASE_URL + "&videoId=" + postRequestBodyData + "&maxResults=" + NUM_RESULTS + "&key=" + KEY);
     String output = get(completeUrl);
     response.setContentType("application/json");
     response.getWriter().println(output);  
   }
   
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    postRequestBodyData = request.getReader().readLine().replaceAll("^\"+|\"+$", "");
+    doGet(request, response);
```

Comment:
Is is not possible to move the code from doGet here? I don't see a need to make an extra request to the server.

Reply:
+1

Reply:
Unfortunately I have not had success with doing that. The YouTube data API will only return a response if a GET response is made to it. My current implementation uses a POST request to update a variable and then makes a GET request with the new information. Although this is not a POST request directly to the YouTube data API it is more secure than a traditional get request.

Reply:
I think Daniel and I had envisioned the solution below.
![image](https://user-images.githubusercontent.com/65875732/89548184-b486e780-d7d4-11ea-9416-34461e73463e.png)
I tested this out on all YouTube servlets and changed the GET requests in JS to POST, and it works.

Reply:
I think Daniel and I had just envisioned the solution below:
![image](https://user-images.githubusercontent.com/65875732/89548419-0465ae80-d7d5-11ea-81b5-b44008674785.png)
I made this change on all YouTube servlets and change `GET` to `POST` in the JS, and it works

Reply:
I think Daniel and I had just envisioned the solution below:
![image](https://user-images.githubusercontent.com/65875732/89548419-0465ae80-d7d5-11ea-81b5-b44008674785.png)
I made this change on all YouTube servlets and change `GET` to `POST` in youtubeScript.js, and it works.

Just a side note, I noticed some code repetition in youtubeScript.js from script.js. If you add script.js in youtube_analyzer.html, we can avoid this

Reply:
+1 to Chris' refactoring

Reply:
+1 to this refactoring.  Just to clarify a bit here, I think we can still use `YoutubeCaller.get` from within this `doPost` method.  Removing `doGet` will just make it clear that a POST should be made to the `/youtube_servlet`

Reply:
I've made this change in "optimize doPost"